### PR TITLE
URI encode the parameters to accept complex value

### PR DESCRIPTION
**Pull request by philippemiguet**
URI encode the parameters to accept complex value like URLs with its own parameters: it will be useful when we will redirect visitors trying to start a trial to the login service. The URI will contain the URL to the start trial endpoint to hit with its own params, like:

`https://buffer.com/billing/start-trial?plan=small&utm_campaign=business-pricing-page&csrf_token=<csrf>&funnel_step_id=6fa6081b-3993-41a3-a080-17a2cbc93360`

### CODE REVIEW

**Comment by hharnisc** [comment]
Test for utils file is here, but doesn't test that function (yet) :smile: : https://github.com/bufferapp/buffer-login/blob/master/test/utils.test.js

**Comment by philippemiguet** [comment]
@esclapes - up for a review? This is an old PR that was missing tests to get merged. Would love your valuable insights on it :)

File: test/utils.test.js [hunk 1]
```diff
@@ -1,10 +1,34 @@
...
   translateCodeToErrorMessage,
 } = require('../src/utils')
 
 describe('utils', () => {
+  describe('getFilteredQueryParams', () => {
+    it('should export getFilteredQueryParams function', () => {
+      expect(getFilteredQueryParams).toBeDefined()
+    })
+
+    it('should encore url with no params', () => {
```
Comment (esclapes) [inline test/utils.test.js]: your French touch-typing took over here with _encore_ :)

File: test/utils.test.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
 const {
+  getFilteredQueryParams,
   validateRedirect,
   translateErrorToMessageCode,
   translateCodeToErrorMessage,
 } = require('../src/utils')
 
 describe('utils', () => {
+  describe('getFilteredQueryParams', () => {
```
Comment (esclapes) [inline test/utils.test.js]: I love seeing tests for `utils`  :raised_hands:   

Would it make sense to add two extra tests to cover these extra cases?

* when no params are passed, it should return an empty string
* when more that one param is passed the result will be a string joining both params with `&`

File: test/utils.test.js [hunk 3]
```diff
@@ -1,10 +1,34 @@
...
+      expect(getFilteredQueryParams).toBeDefined()
+    })
+
+    it('should encore url with no params', () => {
+      const expected = 'redirect=https%3A%2F%2Fpublish.buffer.com%2F'
+      const queryParams = { redirect: 'https://publish.buffer.com/' }
+      const result = getFilteredQueryParams({ queryParams })
+
+      expect(result).toEqual(expected)
+      console.info(getFilteredQueryParams({ queryParams : {redirect: 'https://publish.buffer.com?foo=bar&foo2=bar2'}}))
```
Comment (esclapes) [inline test/utils.test.js]: console.info here :)

File: test/utils.test.js (hunk 1 above)
Comment (philippemiguet) [reply test/utils.test.js]: Hahaha indeed 😂

File: test/utils.test.js (hunk 2 above)
Comment (philippemiguet) [reply test/utils.test.js]: Will add those thanks!

File: test/utils.test.js (hunk 3 above)
Comment (philippemiguet) [reply test/utils.test.js]: Damn again! Thanks Eduardo! :)

**Comment by philippemiguet** [comment]
@esclapes - I've added the new tests, let me know how it feels now :)

**Pull request merged**